### PR TITLE
Fix memory caching issue when doing .compute_contrast in SecondLevelModel

### PR DESCRIPTION
**Pull request by tbng**
Fix issue #272

### CODE REVIEW

File: nistats/tests/test_second_level_model.py [hunk 1]
```diff
@@ -205,3 +205,44 @@ def test_second_level_model_contrast_computation():
...
+        # passing wrong parameters
+        assert_raises(ValueError, model.compute_contrast, [])
+        assert_raises(ValueError, model.compute_contrast, c1, None, '')
+        assert_raises(ValueError, model.compute_contrast, c1, None, [])
+        assert_raises(ValueError, model.compute_contrast, c1, None, None, '')
+        # check that passing no explicit contrast when the dsign
+        # matrix has morr than one columns raises an error
+        X = pd.DataFrame(np.random.rand(4, 2), columns=['r1', 'r2'])
+        model = model.fit(Y, design_matrix=X)
+        assert_raises(ValueError, model.compute_contrast, None)
```
Comment (bthirion) [inline nistats/tests/test_second_level_model.py]: I think that the test is too long for the added functionality.
You probably want to simplify it.

**Comment by tbng** [comment]
I have simplified the test case following the suggestion.

File: nistats/second_level_model.py [hunk 2]
```diff
@@ -406,8 +410,7 @@ def compute_contrast(
         # Fit an OLS regression for parametric statistics
         Y = self.masker_.transform(effect_maps)
         if self.memory is not None:
-            arg_ignore = ['n_jobs']
-            mem_glm = self.memory.cache(run_glm, ignore=arg_ignore)
+            mem_glm = self.memory.cache(run_glm, ignore=['n_jobs'])
```
Comment (kchawla-pi) [inline nistats/second_level_model.py]: You can simply write
```
if self.memory:
```

**Comment by kchawla-pi** [comment]
Nice work @tbng pretty good code quality

File: nistats/second_level_model.py (hunk 2 above)
Comment (tbng) [reply nistats/second_level_model.py]: I agree, I also make the same edit in https://github.com/nistats/nistats/blob/master/nistats/first_level_model.py

**Comment by bthirion** [comment]
LGTM.

**Comment by kchawla-pi** [comment]
Does this need  tests?

**Comment by bthirion** [comment]
> Does this need tests?

Yes, for the first-level model part.

**Comment by kchawla-pi** [comment]
@tbng Clearly, there was a gap in the tests and this malfunction went unnoticed. Please add the necessary tests to this.

**Comment by kchawla-pi** [comment]
@tbng Clearly, there was a gap in the tests and this malfunction went unnoticed. Please add the necessary tests to this.

**Comment by tbng** [comment]
> @tbng Clearly, there was a gap in the tests and this malfunction went unnoticed. Please add the necessary tests to this.

Roger, I can do this.

**Comment by tbng** [comment]
Much apologies for the long delay, I have added the test case for memory caching in first_level_model.

**Comment by kchawla-pi** [comment]
@tbng If you would address the 3 things I pointed out, I can merge this.

**Pull request merged**